cmd/goal/clerk.go

```diff
@@ -425,3 +481,76 @@ var signCmd = &cobra.Command{
 		}
 	},
 }
+
+func assembleFile(fname string) (program []byte) {
+	fin, err := os.Open(fname)
+	if err != nil {
+		reportErrorf("%s: %s\n", fname, err)
+	}
+	ops := logic.OpStream{}
+	err = ops.Assemble(fin)
+	if err != nil {
+		reportErrorf("%s: %s\n", fname, err)
+	}
+	fin.Close()
```

Comment:
I was going to write that I think that you should test the return code of the `Close` function.
But thinking about it, I think that you should use `ioutil.ReadFile` instead, since it would get you a string in one shot. than, you could use the existing AssembeString().